### DISCUSSION TITLE
Minimal steps to add a new dependency

### DISCUSSION THREAD

**Discussion by nkiesel** [Splitties/refreshVersions#525] [General]
Hello,
assuming I have version 0.40.1 setup and working, here is my current recipe for adding a new dependency:

1. make sure the dependency does not already exist by searching in `Libs.kt`
2. add the new dependency as a "literal dependency" to `build.gradle` (or `build.gradle.kts`). Example: add `implementation("org.assertj:assertj-core:3.22.0")`
3. run `./gradlew  buildSrcLibs` which will add a line to `Libs.kt` and a line to `versions.properties` for the new dependency
4. replace the line added in step 2 to `build.gradle` (or `build.gradle.kts`) with `implementation(Libs.assertj_core)`

Is that correct?

**Comment by jmfayard** [comment]
Yes that's correct.

If your library is part of the built-in dependency notations from https://github.com/jmfayard/refreshVersions/tree/main/plugins/dependencies/src/main/kotlin/dependencies
then you would only need to add `implementation(Testing.assertj.core)` 
but assertj is not one of the built-in dependencies

**Reply by nkiesel** [reply]
Thanks! It gets a bit more complicated when adding multiple dependencies covered by a rule (e.g. `assertj-core` and `assertj-db`). I will experiment a bit and then share my recipe for that as well.
